tasks/tasks/scale_indicator.py

```diff
@@ -0,0 +1,76 @@
+from utils import download_rawfile, put_rawfile
+import pandas as pd
+import numpy as np
+import os
+import logging
+
+def scale_indicator(context):
+    logging.info(f"Scaling indicator {context.get('uuid')}")
+    data_paths=context.get('dataset').get('data_paths')
+    min_max_mapping={}
+    for path in data_paths:
+        # current_df=pd.read_parquet(path)
+        filename=path.split('/')[-1]
+        try:        
+            logging.info(f'downloading for s3 {path}')
+            download_rawfile(path, f"processing/{filename}")
+        except FileNotFoundError as e:
+            return {"success":False,'message':"File not found"}
+        
+        current_df=pd.read_parquet(f"processing/{filename}")
+
+        features = current_df.feature.unique()
+        for f in features:
+            feat = current_df[current_df["feature"] == f].copy()
+            current_min=np.min(feat["value"])
+            current_max=np.max(feat["value"])
+            if f not in min_max_mapping:
+                min_max_mapping[f]={"min":current_min, "max":current_max}
+            else:
+                if min_max_mapping[f]["min"] > current_min:
+                    min_max_mapping[f]["min"]=current_min
+                if min_max_mapping[f]["max"]<current_max:
+                    min_max_mapping[f]["max"]=current_max
+    
+    logging.info(f'min max values for all files: {min_max_mapping}')
+
+    for path in data_paths:
```

Comment:
We should check to see if min/max has changed and only overwrite the existing files if they have, otherwise we can simply save the new/missing normalized file(s).

Reply:
Do we save min max values for features anywhere?
I am not aware that we will know which files are new additions or ones that have already been scaled with global min max. If we save current min max in elasticsearch I can see this working and its probably a good idea to store that, but I am not seeing those values.

Reply:
New additions are files that exist in data_paths that do not (yet) have a corresponding _normalized entry.
We can calculate the min/max of the original files first, then the additions, and if the min max doesn't change then we still have to write the normalized files for the new additions, but don't have to do any work on the existing normalized files since there would be no change.

Reply:
Ok so currently my script normalizes the datasets and saves them in s3. I will just need to remove that code and modify the min_max mapping to compare the differences between the old verse new files.